src/test/java/com/amazon/opendistroforelasticsearch/sql/unittest/WhereWithBoolConditionTest.java

```diff
@@ -0,0 +1,71 @@
+/*
+ *   Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License").
+ *   You may not use this file except in compliance with the License.
+ *   A copy of the License is located at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   or in the "license" file accompanying this file. This file is distributed
+ *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *   express or implied. See the License for the specific language governing
+ *   permissions and limitations under the License.
+ */
+
+package com.amazon.opendistroforelasticsearch.sql.unittest;
+
+import com.alibaba.druid.sql.parser.ParserException;
+import com.amazon.opendistroforelasticsearch.sql.esintgtest.TestsConstants;
+import com.amazon.opendistroforelasticsearch.sql.exception.SqlParseException;
+import com.amazon.opendistroforelasticsearch.sql.query.ESActionFactory;
+import com.amazon.opendistroforelasticsearch.sql.query.QueryAction;
+import com.amazon.opendistroforelasticsearch.sql.util.CheckScriptContents;
+import org.elasticsearch.client.Client;
+import org.junit.Test;
+import org.mockito.Mockito;
+
+import java.sql.SQLFeatureNotSupportedException;
+
+import static org.junit.Assert.assertTrue;
+
+
+public class WhereWithBoolConditionTest {
+
+    private static final String SELECT_ALL = "SELECT * ";
+    private static final String SELECT_ADDRESS = "SELECT address ";
+    private static final String FROM_BANK = "FROM " + TestsConstants.TEST_INDEX_BANK;
+    private static final String ORDER_BY_AGE = "ORDER BY age ";
+    private static final String GROUP_BY_BALANCE = "GROUP BY balance ";
+    private static final String WHERE_CONDITION_TRUE = "WHERE male = true";     // 4 hits
+    private static final String WHERE_CONDITION_FALSE = "WHERE male = false";   // 3 hits
+
+    @Test
+    public void whereWithBoolCompilationTest() {
+       query(FROM_BANK, WHERE_CONDITION_FALSE);
+    }
+
+    @Test
+    public void selectAllTest() {
+        String matchQuery = "{\"from\":0,\"size\":200," +
+                "\"query\":{\"bool\":{\"filter\":[{\"bool\":{\"must\":" +
+                "[{\"term\":{\"male\":{\"value\":true,\"boost\":1.0}}}],\"adjust_pure_negative\":true,\"boost\":1.0}}],\"adjust_pure_negative\":true,\"boost\":1.0}}}";
+
+        assertTrue(query(FROM_BANK, WHERE_CONDITION_TRUE).equals(matchQuery));
```

Comment:
To make it straightforward, I think you can put your query here directly rather than combine its pieces to one.

Reply:
My main reason for separating it was readability- if it's combined into 1 the line becomes extremely long, is that still preferable? If so, I'll make the change

Reply:
Yes, please, make it inline. As a reader I have no idea what's going on here